src/anyp/TrafficMode.h

```diff
@@ -58,27 +67,99 @@ class TrafficMode
      *  - Same-Origin verification is mandatory
      *  - destination pinning is recommended
-     *  - authentication prohibited
+     *  - Squid authentication prohibited
      */
     bool tproxyIntercept = false;
+};
+
+/**
+ * Set of 'mode' flags defining types of traffic which can be received.
+ *
+ * Use to determine the processing steps which need to be applied
+ * to this traffic under any special circumstances which may apply.
+ */
+class TrafficMode
+{
+public:
+    /// This port handles traffic that has been intercepted prior to being delivered
+    /// to the TCP client of the accepted connection and/or to us. This port mode
+    /// alone does not imply that the client of the accepted TCP connection was not
+    /// connecting directly to this port (since commit 151ba0d).
+    bool interceptedSomewhere() const { return flags_.natIntercept || flags_.tproxyIntercept || flags_.proxySurrogateHttps; }
+
+    /// The client of the accepted TCP connection was connecting to this port.
+    /// The accepted traffic may have been intercepted earlier!
+    bool tcpToUs() const { return proxySurrogate() || !interceptedSomewhere(); }
+
+    /// The client of the accepted TCP connection was not connecting to this port.
+    /// The accepted traffic may have been intercepted earlier as well!
+    bool interceptedLocally() const { return interceptedSomewhere() && !tcpToUs(); }
+
+    // Unused yet.
+    /// This port handles traffic that has been intercepted prior to being delivered
+    /// to the TCP client of the accepted connection (which then connected to us).
+    bool interceptedRemotely() const { return interceptedSomewhere() && tcpToUs(); }
+
+    /// The client of the accepted TCP connection was connecting directly to this proxy port.
+    bool forwarded() const { return !interceptedSomewhere() && !flags_.accelSurrogate; }
+
+    /// whether the PROXY protocol header is required
+    bool proxySurrogate() const { return flags_.proxySurrogateHttp || flags_.proxySurrogateHttps; }
+
+    /// interceptedLocally() with configured NAT interception
+    bool natInterceptLocally() const { return flags_.natIntercept && interceptedLocally(); }
+
+    /// interceptedLocally() with configured TPROXY interception
+    bool tproxyInterceptLocally() const { return flags_.tproxyIntercept && interceptedLocally(); }
+
+    /// whether the reverse proxy is configured
+    bool accelSurrogate() const { return flags_.accelSurrogate; }
+
+    TrafficModeFlags &rawConfig() { return flags_; }
+
+    std::ostream &print(std::ostream &) const;
 
     /** marks intercept and decryption of CONNECT (tunnel) SSL traffic
      *
      * Indicating the following are required:
      *  - decryption of CONNECT request
      *  - URL translation from relative to absolute form
-     *  - authentication prohibited on unwrapped requests (only on the CONNECT tunnel)
+     *  - Squid authentication prohibited on unwrapped requests (only on the CONNECT tunnel)
```

Comment:
Still relevant

Reply:
Since we [cannot](https://wiki.squid-cache.org/SquidFaq/InterceptionProxy#Why_can.27t_I_use_authentication_together_with_interception_proxying.3F) use authentication with any interception, I undid (762cd28) this change.